packages/vats/test/bootstrapTests/supports.js

```diff
@@ -67,42 +67,51 @@ export const makeRunUtils = (controller, log = (..._) => {}) => {
 
   mutex.put(controller.run());
 
-  const runMethod = async (method, args = []) => {
-    log('runMethod', method, args, 'at', cranksRun);
-    assert(Array.isArray(args));
-
+  /**
+   * @template {() => any} T
+   * @param {T} thunk
+   * @returns {Promise<ReturnType<T>>}
+   */
+  const runThunk = async thunk => {
     try {
       // this promise for the last lock may fail
       await mutex.get();
     } catch {
       // noop because the result will resolve for the previous runMethod return
     }
 
-    const kpid = controller.queueToVatRoot('bootstrap', method, args);
-
-    const getResult = () => {
-      const status = controller.kpStatus(kpid);
-      switch (status) {
-        case 'fulfilled':
-          return kunser(controller.kpResolution(kpid));
-        case 'rejected':
-          throw kunser(controller.kpResolution(kpid));
-        case 'unresolved':
-          throw Error(`unresolved for method ${method}`);
-        default:
-          throw Fail`unknown status ${status}`;
-      }
-    };
+    const thunkResult = await thunk();
```

Comment:
Note that awaiting introduces a difference in behavior where `controller.run` will not execute until the thunk has completed, so I don't see how the `runMethod` refactor would actually work.

Reply:
As before, it's doing some work to prepare for the run(), waiting the run() and then doing some work to get something useful out of the result. The tests are still passing so I'm sufficiently confident in this to merge.

Reply:
I misread `runMethod`. The `getResult` part is not done inside the `thunk`. However that means it's done after the mutex is released. I assumed it's safe enough to run `controller.kpStatus` and `controller.kpResolution` while another `controller.run` might have started?

Reply:
Yes I think so because they're using the same unique `kpid`.